components/fxa-client/ios/FxAClient/RustFxAccount.swift

```diff
@@ -104,18 +104,48 @@ open class RustFxAccount {
     /// Once the user has confirmed the authorization grant, they will get redirected to `redirect_url`:
     /// the caller must intercept that redirection, extract the `code` and `state` query parameters and call
     /// `completeOAuthFlow(...)` to complete the flow.
-    open func beginOAuthFlow(scopes: [String], entrypoint: String) throws -> URL {
+    open func beginOAuthFlow(
+        scopes: [String],
+        entrypoint: String,
+        metricsParams: MetricsParams
+    ) throws -> URL {
         let scope = scopes.joined(separator: " ")
-        let ptr = try rustCall { err in
-            fxa_begin_oauth_flow(self.raw, scope, entrypoint, err)
+        let (metricsParamsData, metricsParamsLen) = msgToBuffer(msg: metricsParams.toMsg())
+        let ptr = try metricsParamsData.withUnsafeBytes { bytes in
+            try rustCall { err in
+                fxa_begin_oauth_flow(
+                    self.raw,
+                    scope,
+                    entrypoint,
+                    bytes.bindMemory(to: UInt8.self).baseAddress!,
```

Comment:
Honestly not sure what this does, but what was done in the other apis.

Reply:
me neither :-/